src/CHIPBindings.cc

```diff
@@ -3622,6 +3632,8 @@ hipError_t hipMemcpyHtoA(hipArray *DstArray, size_t DstOffset,
 hipError_t hipMemcpy3DAsyncInternal(const struct hipMemcpy3DParms *Params,
                                     hipStream_t Stream) {
   auto ChipQueue = Backend->findQueue(static_cast<chipstar::Queue *>(Stream));
+  LOCK(ChipQueue->QueueMtx); // prevent interruptions
+
   if (ChipQueue->captureIntoGraph<CHIPGraphNodeMemcpy>(Params)) {
     return hipSuccess;
```

Comment:
UNLOCK for QueueMtx?? Perhaps lock guard?

Reply:
LOCK is a macro which creates a lock guard